### DISCUSSION TITLE
Keka v1.2.61 seems to be creating .zip files which are incompatible with WinZip 25.0?

### DISCUSSION THREAD

**Discussion by rstewart-inl** [aonez/Keka#1265] [Q&A]
I've used Keka for years to produce .zip files to share with people using WinZip on Windows systems without any issues. In the last week, I've bumped into two different people using WinZip 25.0 who can't seem to open different Keka-produced .zip files. In both cases, these are people who've not encountered this in the past (I would have been on an earlier version of Keka and they may have been on earlier versions of WinZip). I've verified the Keka-produced .zip files are usable with WinZip 22.5 (the only version I have access to).

Is this something I need to differently? ... or a bug of some sort in Keka? ... or in WinZip?

Ideas definitely welcome, and thanks in advance.

**Comment by aonez** [comment]
I'll check tomorrow that WinZip version. Nothing has changed in the Zip creation on Keka.

Let me know the specifics of those files if any, such if they are password protected with or without AES256, size or anything that can help reproduce the issue.

Also sharing a file that reproduces the issue will be ideal, you can share it here or at info@keka.👍🏼

**Comment by rstewart-inl** [comment]
They aren't password-protected, and size doesn't seem to matter. They are compressed with Method "slowest", not split, exclude Mac resource forks.

One was quite small (24KB), the other was larger (~6MB). I've reverted one of my Macs to v1.2.60 and sent a .zip to one of these two people created with the same content to see if that shows the same behavior. I'll work on getting you a test file.

**Comment by rstewart-inl** [comment]
@aonez Both of the attached test files apparently exhibit this behavior with WinZip 25.0 Enterprise (14273). I've had them try this with a .zip created with Keka v1.2.60, as well, and WinZip seems to fail with it, as well. 

[lipsum-1.2.61.zip](https://github.com/aonez/Keka/files/11559585/lipsum-1.2.61.zip)
[lipsum-1.2.60.zip](https://github.com/aonez/Keka/files/11559586/lipsum-1.2.60.zip)

**Comment by aonez** [comment]
Just extracted them without any issue. Also tested a bunch of other files. Is it possible that the files were corrupted when being shared?

You can check the checksum, for example `MD5`.

- On macOS, using the Terminal: `md5 /path/to/file.zip`
- On Windows, using CMD or PowerShell: `certutil -hashfile \path\to\file.zip MD5`

I've tested with the latest version of WinZip which seems to be 27.0.

<img width="937" alt="Screenshot 2023-05-25 at 00 38 50" src="https://github.com/aonez/Keka/assets/480163/8a36dc5d-8a84-4b9e-b64c-cb0b1cbf14ba">

**Reply by aonez** [reply]
They also can be opened and extracted with the File Explorer of Windows.

**Comment by rstewart-inl** [comment]
I had them verify the checksum in both cases and their checksums matched mine, so I don't believe they were corrupted. In testing those lipsum files, I also sent a .7z version of the same file -- compressed with Keka v1.2.61 -- and they were able to uncompress it successfully with WinZip 25.0.

This is starting to feel like something specific to that version of WinZip, since your v27 works and my v22.5 works but two people with v25 seem to be failing?

And thanks for the quick follow-up on this, @aonez . Much appreciated.

**Comment by aonez** [comment]
Can they open those files with the File Explorer?

Out of curiosity found the installer of WinZip 25.0 and tried it without issues too. Note those are not the Enterprise edition as I did not find the installer of that version.

<img width="1051" alt="Screenshot 2023-05-25 at 00 58 03" src="https://github.com/aonez/Keka/assets/480163/0ca6df43-9ae3-4d14-8162-6a200bf17c86">

**Comment by rstewart-inl** [comment]
Confirmed they can open them without issue using just Windows File Explorer. This is baffling. I'd chalk it off to something being goofy with their WinZip installation but these are two individuals who work for entirely different organizations in different parts of the country.

This seems clear to me that it is not a problem on the creating (Keka) end of this, and there's an obvious work-around via File Explorer.

Thanks for your time in digging into this and trying the different versions of WinZip, @aonez
